pkg/controller/plan/kubevirt.go

```diff
@@ -1644,7 +1650,7 @@ func vmOwnerReference(vm *cnv.VirtualMachine) (ref meta.OwnerReference) {
 
 // TODO move elsewhere
 func (r *KubeVirt) createOpenStackVolumes(vm ref.Ref) (err error) {
-	secret, err := r.ensureSecret(vm, false)
+	secret, err := r.ensureSecret(vm, r.secretDataSetterForCDI(vm))
```

Comment:
CDI doesn't support OSP. but we have singular usage of the builder. that's why it was ok with false.

Reply:
sure, it's not that it didn't work - but it was misleading, we actually do want to clone the provider secret for openstack as well, now that the code is more explicit we can see that